wcfsetup/install/files/lib/system/session/SessionHandler.class.php

```diff
@@ -405,46 +370,114 @@ public function getUser() {
 	}
 	
 	/**
-	 * Tries to read existing session identified by the given session id.
-	 * 
-	 * @param	string		$sessionID
+	 * Tries to read existing session identified by the given session id. Returns whether
+	 * a session could be found.
 	 */
-	protected function getExistingSession($sessionID) {
-		$this->session = new $this->sessionClassName($sessionID);
-		if (!$this->session->sessionID) {
-			$this->session = null;
-			return;
+	protected function getExistingSession(string $sessionID): bool {
+		$sql = "SELECT	*
+			FROM	wcf".WCF_N."_".($this->isACP ? 'acp' : 'user')."_session
+			WHERE	sessionID = ?";
+		$statement = WCF::getDB()->prepareStatement($sql);
+		$statement->execute([
+			$sessionID
+		]);
+		$row = $statement->fetchSingleRow();
+		
+		if (!$row) {
+			return false;
+		}
+		
+		// Check whether the session technically already expired.
+		$lifetime = ($this->isACP ? self::ACP_SESSION_LIFETIME :
+			($row['userID'] ? self::USER_SESSION_LIFETIME :
+			self::GUEST_SESSION_LIFETIME));
+		if ($row['lastActivityTime'] < (TIME_NOW - $lifetime)) {
+			return false;
+		}
+		
+		$this->sessionID = $sessionID;
+		$this->user = new User($row['userID']);
+		$this->variables = unserialize($row['sessionVariables']);
```

Comment:
```suggestion
		$this->variables = @unserialize($row['sessionVariables']);
		if (!is_array($this->variables)) $this->variables = [];
```

Rationale: Serialized values become magically corrupted in shared hosting environments, better discard the value than bricking this session.

Reply:
I honestly don't see this happening. Especially since the `sessionVariables` are now properly stored in a `BLOB` column. But I'd rather drop the whole session instead of pretending everything is fine, even if it is not.